### PR TITLE
Plonts for condiments

### PR DESCRIPTION
**Pull request by xXPawnStarrXx**
## About The Pull Request
I noticed that with the cooking update that chefs were having issues sourcing good quality salt and pepper.
Introducing Pepper-corn and Saltcane! They are available as mutations of corn and sugarcane, so you better have a green thumb!
(I'm aware peppercorns are a thing; that's the joke!)
image

EDIT: Added a darker green sprite to the saltcane to make it more distinguishable from sugarcane as well as it's sprite orientation.
277135378-95019f25-049b-4fc5-9c3b-616881a813e4
image

## Why It's Good For The Game
Makes getting reagents like pepper for cooking with a quality above 0.3 actually acheivable for chefs without them hemorrhaging money if they have a capable botanist. Benefitting both lizards and those that can't live without McRibs.
## Changelog
:cl:
add: Added new plants for gathering salt and pepper.
/:cl:

### CODE REVIEW

**Comment by xXPawnStarrXx** [comment]
Let me know if you think anything needs to be changed or adjusted. I want to make sure that it's all good.

**Comment by optimumtact** [comment]
Bro got me thinking plont was some gen z slang i'm not up on.

**Comment by xXPawnStarrXx** [comment]
> Bro got me thinking plont was some gen z slang i'm not up on.

Nah, I typoed and it was hilarious so I kept it in.

**Comment by the-orange-cow** [comment]
Just a thought:  both of these will be branched mutations, which are inherently higher maintenance to obtain without advanced botany tools.  If this PR is primarily meant for convenience, the branched mutation aspect will probably make it borderline inaccessible for most chefs, and even a lot of botanists might be less interested in the effort for condiments.

A possibly alternative:  have pepper be obtainable through grinding chili, and salt obtained through something like seaweed maybe.

Again, only mentioning this because branched mutations might be a slightly higher hurdle than something like salt and pepper deserve for how mundane they are.

**Comment by xXPawnStarrXx** [comment]
> Again, only mentioning this because branched mutations might be a slightly higher hurdle than something like salt and pepper deserve for how mundane they are.

I didn't want to outright remove the value of just being able to order high purity salt and pepper from cargo/service console and also wanted it so botanists could grow 1-2 harvests and at 20/25% content respectively, should last a fair amount of time. I wanted the need for a botanist to be a thing, otherwise they can spend more and get it from cargo.

**Comment by Treachce** [comment]
I thought the biogenerator could make pepper? Regardless, more plants and another way to get salt is nice.

**Comment by xXPawnStarrXx** [comment]
> I thought the biogenerator could make pepper? Regardless, more plants and another way to get salt is nice.

it can, however with recent changes it's purity 0.3. Plants are purity 0.8 and cargo bought stuff is all purity 1.0 this is to branch the disparity between 0.3 and 1.0

**Comment by CRITAWAKETS** [comment]
> it can, however with recent changes it's purity 0.3. Plants are purity 0.8 and cargo bought stuff is all purity 1.0 this is to branch the disparity between 0.3 and 1.0

Plants have a purity algorithm so they can go from ~27% to 100%, depending on if your botanist is Dr. Dumbass or the equivalent of an Ace for botanists. Generally they'll be somewhat above 50-60% though.

**Comment by Wallemations** [comment]
I'm not a botany main, is the salt cane visually distinct enough from sugarcane?

**Comment by xXPawnStarrXx** [comment]
> I'm not a botany main, is the salt cane visually distinct enough from sugarcane?
> 
> Same goes for the stalk of the corn

The sugarcane and salt cane look the same when growing, however the sprite is rotated after harvest. (I tried to change the colours to more of a white inside but unless compared under a microscope, it didn't change much.)

The peppercorn is visible when fullygrown as it has a little black head.

**Comment by Wallemations** [comment]
> The sugarcane and salt cane look the same when growing, however the sprite is rotated after harvest.

I don't think you should just use the same sprite, at least change the color a little so it's somewhat visible. It doesn't matter if it's just a little more yellow or green, or even greyscaled, just as long as it's a visible difference.

**Comment by xXPawnStarrXx** [comment]
> > The sugarcane and salt cane look the same when growing, however the sprite is rotated after harvest.
> 
> I don't think you should just use the same sprite, at least change the color a little so it's somewhat visible. It doesn't matter if it's just a little more yellow or green, or even greyscaled, just as long as it's a visible difference.

I actually tried to change the interior colour but it's so pale already that the white doesn't even stand out. I can give it a second shot it you want.

**Comment by Wallemations** [comment]
Not just the item sprite, the harvest sprite

**Comment by xXPawnStarrXx** [comment]
> Not just the item sprite, the harvest sprite

Donezo
image

**Comment by xXPawnStarrXx** [comment]
I know it's failed. I'll check when I wake up and try fix it.

**Review by Wallemations — APPROVED** [review]
Better

File: code/modules/hydroponics/grown/corn.dm [hunk 1]
```diff
@@ -97,3 +97,28 @@
...
+
+/obj/item/food/grown/peppercorn
+	seed = /obj/item/seeds/corn/pepper
+	name = "ear of pepper-peppercorn"
+	desc = "This dusty monster needs god..."
+	icon_state = "peppercorn"
+	trash_type = /obj/item/grown/corncob/pepper
+	foodtypes = VEGETABLES
+	grind_results = list(/datum/reagent/consumable/blackpepper = 0)
+	tastes = list("pepper, sneezing" = 1)
```
Comment (MrMelbert) [inline code/modules/hydroponics/grown/corn.dm:119]:
```suggestion
	tastes = list("pepper" = 1, "sneezing" = 1)
```
When tasted, it'll end up being `Tastes like pepper and sneezing` or `Tastes like pepper, sneezing, etc`. So no point to combine them.

**Review by MrMelbert — APPROVED** [review]

**Comment by MrMelbert** [comment]
Whoops. I self-commited that without realizing there's merge conflicts, my bad.

**Comment by xXPawnStarrXx** [comment]
> Whoops. I self-commited that without realizing there's merge conflicts, my bad.

That's fine. I am not at home until later tomorrow, so I can't see what's conflicting until after then.

**Comment by xXPawnStarrXx** [comment]
I messed up pretty badly, I'm just going to start over with a new fork.

**Pull request closed without merging**